gnovm/pkg/gnolang/preprocess.go

```diff
@@ -5497,3 +5500,184 @@ func SaveBlockNodes(store Store, fn *FileNode) {
+							validateMakeArg(store, last, currExpr)
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+// Validate argument for len() function
+func validateLenArg(at Type) {
+	if at == nil {
+		panic("invalid argument: nil for built-in len")
+	}
+
+	switch at := unwrapPointerType(baseOf(at)).(type) {
+	case PrimitiveType:
+		if at.Kind() != StringKind && at != UntypedStringType {
+			panic(fmt.Sprintf("unexpected type for len(): %v", at))
+		}
+	case *ArrayType, *SliceType, *MapType, *ChanType:
+		// Valid types for len()
+	default:
+		panic(fmt.Sprintf("unexpected type for len(): %v", at))
+	}
+}
+
+// Validate argument for cap() function
+func validateCapArg(at Type) {
+	if at == nil {
+		panic("invalid argument: nil for built-in cap")
+	}
+
+	switch at := unwrapPointerType(baseOf(at)).(type) {
+	case *ArrayType, *SliceType, *ChanType:
+		// Valid types for cap()
+	default:
+		panic(fmt.Sprintf("unexpected type for cap(): %v", at))
+	}
+}
+
+// Validate arguments for make() function
+func validateMakeArg(store Store, last BlockNode, currExpr *CallExpr) {
+	if len(currExpr.Args) < 1 || len(currExpr.Args) > 3 {
+		panic(fmt.Sprintf("invalid number of arguments for built-in make: expected 1 to 3, got %d", len(currExpr.Args)))
+	}
+
+	// Validate first argument: must be a slice, map, or channel type
+	at := evalStaticType(store, last, currExpr.Args[0])
+	validateMakeType(baseOf(at))
+
+	var length int
+	var capacity int
+
+	// Validate length argument (second argument for make)
+	if len(currExpr.Args) > 1 {
+		at1 := evalStaticTypeOf(store, last, currExpr.Args[1])
+		if at1 == nil {
+			panic("invalid argument 2: nil for built-in make")
+		}
+
+		var err error
+		length, err = getIntValue(currExpr.Args[1], "length")
+		if err != nil {
+			panic(err.Error()) // Convert error to panic message
+		}
+		if length < 0 {
+			panic(fmt.Sprintf("invalid length argument for built-in %s: cannot be negative, got %d", BuiltinMake, length))
+		}
+	}
+
+	// Validate capacity argument (third argument for make)
+	if len(currExpr.Args) > 2 {
+		at2 := evalStaticTypeOf(store, last, currExpr.Args[2])
+		if at2 == nil {
+			panic("invalid argument 3: nil for built-in make")
+		}
+		var err error
+		capacity, err = getIntValue(currExpr.Args[2], "capacity")
+		if err != nil {
+			panic(err.Error()) // Convert error to panic message
+		}
+		if capacity < length {
+			capacity = length
+		}
+	}
+}
+
+// Ensures that the first argument to make() is a valid type
+func validateMakeType(tp Type) {
+	switch tp.(type) {
+	case *SliceType, *MapType, *ChanType:
+		// Valid types for make()
+	default:
+		panic(fmt.Sprintf("invalid type for built-in %s: %s (must be slice, map, or channel)", BuiltinMake, tp.String()))
+	}
+}
+
+func getIntValue(expr Expr, argName string) (int, error) {
+	switch e := expr.(type) {
+	case *ConstExpr:
+		return getIntValue(e.Source, argName)
+
+	case *BasicLitExpr:
+		var num int64
+		if e.Kind == FLOAT {
+			f, err := hasFractionalPart(e.Value)
```

Comment:
Can you change the name of this function? The name looks like it returns a boolean but actually returns a number and an error.